cloud/google/storage.py

```diff
@@ -15,10 +18,35 @@ def download(self, bucket_name: str, source_filename: str, destination_filename:
         blob = self.client.bucket(bucket_name).blob(source_filename)
         blob.download_to_filename(destination_filename)
 
+    def generate_presigned_url(
+        self,
+        bucket_name: str,
+        source_filename: str,
+        expiration: int = 3600,
+    ) -> str:
+        credentials, _ = auth.default()
+        if credentials.token is None:
+            credentials.refresh(requests.Request())
+
+        blob = self.client.bucket(bucket_name).blob(source_filename)
+
+        url = blob.generate_signed_url(
+            version="v4",
+            service_account_email=credentials.service_account_email,
+            access_token=credentials.token,
```

Comment:
Se não for possível inicializar o cliente com as credenciais vamos passar as credenciais aqui direto, nela já estão as informações necessárias e não deveríamos passar separadamente token e email.